### DISCUSSION TITLE
v18 has built-in fetch()

### DISCUSSION THREAD

**Discussion by thetayloredman** [node-fetch/node-fetch#1537] [Announcements]
Expeimental [`fetch()`](https://nodejs.org/en/blog/announcements/v18-release-announce/#fetch-experimental) support is now in `node:fetch` as of v18. Is this the end of node-fetch?

**Comment by AaronDewes** [comment]
Node fetch still has some unique features missing in undici, for example [using a custom agent](https://github.com/node-fetch/node-fetch#custom-agent). Also, it's still useful when trying to support older node versions

**Comment by jimmywarting** [comment]
have to agree that it's useful and nice that it's built in in core now and has support for whatwg streams, it's well written and follow the spec very well. but as with everything else, it's experimental and 2 things node-fetch is better at then undici is parsing formdata payloads and handle larger blobs with streaming. both has pro & cons

with blobs over 500mib `blob.text()` would fail in NodeJS buffer.Blob
that's not the case with `fetch-blob`

**Reply by rafde** [reply]
Also, there's no buffer in the built-in fetch, which is a problem since I can't return the buffer from nextjs api.

I had to do the following to make it work now

```js
const pdf = await response.then(
		( res ) => res.arrayBuffer()
	)
		.then( ( ab ) => Buffer.from( ab ) );
```

**Reply by jimmywarting** [reply]
I do not think one should be using a node Buffer, it's not cross env friendly, it's better to just use Uint8Array to avoid bloating a browser bundle with a buffer polyfill. I have always thought it's a bad idea to extend  uint8array from the start, the `slice` method dose not do the same thing you would expect it to do... hence why we deprecated `.buffer()`. also b/c it's non standard spec compatible

need to read it? use DataView
need to convert to/from text and uint8array? use textencoder/decoder

```js
// btw, i would have written it like this...
const pdf = await response.then(res => res.arrayBuffer()).then(Buffer.from)
```

**Comment by wenq1** [comment]
In nodejs 18 release notes, it is announced that 

“In` Node.js 18, an experimental global fetch API is available by default. The implementation comes from [undici](https://undici.nodejs.org/#/) and is inspired by [node-fetch](https://github.com/node-fetch/node-fetch) which was originally based upon [undici-fetch](https://github.com/Ethan-Arrowood/undici-fetch).”

Is it true that node-fetch was “originally based on undock-fetch” or that the announcement has misphrased?
